### ISSUE TITLE
Plan and fit the excellent networking in the new space

### ISSUE THREAD

**Issue by paulfurley** [DoESLiverpool/somebody-should#598]
For those of us that spend most of our time working remotely with others, the quality of our network connection is a high priority for video calling and collaboration. 

And an excellent tech community deserves an excellent internet connection!

It was clear from pub discussion that I have some strong opinions on what that means!

Here are my suggestions:

## Encourage wired internet

Naturally some devices need to connect via WiFi, but the performance of WiFi degrades exponentially with the number of clients. So it's better for everyone if we encourage wired connections for those devices which support it.

Practically, we should:
- aim to get ethernet to every desk
- help people understand why it's good for everyone to use that
- recommend a USB ethernet dongle thing for fat-free laptop users

## Have a good uplink & download speed

Of course, we need a good internet speed both ways, and we should be sure we're getting what we think we're getting.

Not sure what performance we actually need, but for reference a single device doing HD video will use  ~3.5Mbps.

## Have (at least 1) static IPv4 address

We're going to be doing our own NAT (sadly) and we shouldn't have any more NAT boxes upstream of us. This means we can manage our own port forwarding and run servers and stuff inside DoES.

Also, NAT is evil, so...

## Have IPv6

The UK is embarrassingly behind most of the world on IPv6.

Since DoES is clearly the most interesting tech scene in the city which will apparently soon have [world-class connectivity](http://www.businesscloud.co.uk/news/liverpool-region-can-attract-global-tech-companies), we should be blazing the trail here. (I've had IPv6 at *home* for a few years now...)

## No captive portals, ever!

Captive portals are those awful things which pop up and break your internet connection while you wait for a pointless 'Connect now' button to appear.

They complicate getting things online - particularly internet-of-things things - and they serve absolutely no purpose.

## We can measure and monitor performance

From my experience it's actually really complicated diagnosing *why* a video call doesn't work well. It's really helpful if you've got some data to look back on.

I'm happy to volunteer on this one - we should be a bit pre-emptive about measuring and monitoring our connection and our WiFi clients, channels and throughput so that when we do have problems we can identify whether it's the router, too many clients, unreliable provider, whatever.

## We could try out a city-wide network name

This is bit off the wall and it's harking back to the MerseyNet idea from a few years ago.

Wouldn't it be lovely if we and other places in the city used the same network name (SSID), so you could just roam freely between places?

@amcewen's suggestion is still my fave:
network name: `ourwifi`
password `ourwifi`

Please pitch in and let's work out what we want!

**Comment by zarino** [comment]
Good start! Maybe also:

## Work out the optimum router placement / configuration for good wifi coverage

How many routers, which models, in which positions? Do we have one SSID for maximum ease of use, or multiple SSIDs (á la `DoESLiverpool/DoESDinky`), so we can force devices onto the strongest network in their area of the building?

## Good internet as a selling point for the events / meeting rooms

If we want to encourage repeat customers for the event spaces and meeting rooms, ensuring we have wifi that is easy to connect to, and can support business activities like video calls, will put us ahead of the competition, and could result in more repeat bookings. (I know _my_ organisation is pretty ruthless with finding a new meeting space if the previous one had poor wifi.)

**Comment by livefixtoyou** [comment]
Since I'm new, I'll just throw a couple of suggestions in regard to WiFi quality.

A simple way would be to just put the routers somewhere temporary initially and go round with a mobile phone measuring the wifi strength signal. There's plenty of free apps to measure the signal in dB. Some apps will suggest the best channel to use. You can also try difference devices in different places to get a good idea of average coverage.

The harder way to do it would be using Wireshark on a laptop (overkill?). You'd then be able to isolate specific issues in details. I'll be honest, I've only fired up Wireshark a few times so I'm by no means useful on it. But I can install and set it up to capture packets and protocol handshakes, headers, etc. But it would take someone with a lot more experise than myself to go through what could be saved over time.

Finally, although I don't work in IT, I'm CompTIA A+ and CompTIA Network+ qualified. So I'm willing to help out if needed.

**Comment by johnmckerrell** [comment]
Thought I should add that we have a specific requirement to have wired networking in the events room, the meeting room and the soundproof room. This should be met by "Encourage wired internet" on the off change we change our minds on that.

**Comment by de14n** [comment]
In regards to measuring connectivity issues... has anyone come across Wireshark at all? I've used to capture locally between my laptop and my WiFi, but I'm no expert when it comes to analysing the data afterwards though but it would give some obvious clues to look into and work issues out. It goes without saying that it would need to be configured to capture specific test computer/connections because it's hacker level stuff.

Although I don't work in IT, I am CompTIA certified A+ and Network+ (not quite cisco level though, but very broad nonetheless). So if I can be of any help just give me a shout. I'd be quite happy to help wire up unless it'll be outsourced.

**Comment by xj25vm** [comment]
@de14n - I believe Wireshark comes in handy when troubleshooting networking issues at protocol level. I don't think it is very useful if the problems are caused by speed or latency (or lack off) - as suggested in the original post. I have looked into this for some clients, and there doesn't seem to be an easy answer. To truly have a useful tool, one would need a tool which records and stores traffic flow and content characteristics over time, with high frequency, and high granularity - including all protocols used at a particular point in time, origin IP addresses, destination IP addresses, and size/number of packets for each one of these aspects. This is a large amount of data, and requires quite a bit of processing power. However, I am still on the look-out for a tool which would do the above efficiently - maybe there is one out there :-)

**Comment by xj25vm** [comment]
re: Have a good upload and download link - what type of connection and speed will be available at the new premises (and within budget)? FTTC, FTTP, microwave?

**Comment by ajlennon** [comment]
You can do things with Wireshark but imho it is a pain to get anywhere with it.

A standard install will give you wireless packets for your device with the interesting radio fields stripped off. Depending on the wireless adapter hardware you have and what its drivers support you may be able to put it into a promiscuous mode (so you can see all the packets flying over the air) and get the additional radio information out.

I've done this and whilst I'm relatively au fair with Wireshark (I've written 802.15.4 plugins to decode a proprietary mesh protocol received through a serial gateway->local pipe->wireshark) I still found it fairly incomprehensible to try to work out what was going on.

e.g. https://wiki.wireshark.org/CaptureSetup/WLAN

However there are some decent tools out there I think. I can't recall names off the top of my head but wireless coverage / hand off and so forth is quite an active area so we should be able to dig up something simpler to use than wireshark

For example I've got this thing on my laptop which gives quite a nice overview of the channels the networks around me are sitting on

![image](https://user-images.githubusercontent.com/1537834/32490784-4e1ad284-c3ac-11e7-8309-0990bcc7a2f8.png)

**Comment by xj25vm** [comment]
@ajlennon - I was thinking more along the lines of monitoring network traffic - a software on the gateway, which looks at all incoming and outgoing traffic and stores metadata about it for later analysis - such as which protocols are used, how much bandwidth various applications use, when are the busy times of the day etc.

There are various tools for analysing wifi networks as well - but I found that, in general, once the wifi setup is working properly, there doesn't seem to be much need to monitor that side of things.

**Comment by ajlennon** [comment]
So kind of like what the NSA does with the Utah data centre?

**Comment by xj25vm** [comment]
Err - no - not really. They (presumably) collect actual data. For the purposes of managing a network, it is metadata that is mostly needed. Speed, latency, which applications or type of traffic use bandwidth and when. Although, I accept that there could be some privacy issues involved even when collecting metadata.

**Comment by ajlennon** [comment]
Surely meta-data is data? (There's quite a big discussion about this as relates to privacy...)

**Comment by ajlennon** [comment]
Interesting piece here -

https://opendatasecurity.io/what-is-metadata-and-what-does-it-reveal/

Stallman is as good as ever on surveillance here -

https://www.wired.com/2013/10/a-necessary-evil-what-it-takes-for-democracy-to-survive-surveillance/

I'd be pretty unhappy to be surveilled in this way myself.

**Comment by xj25vm** [comment]
Sorry - I just realised Github is only displaying my username - instead of the full name.

@ajlennon - yes, there could be some privacy issues involved there. I suppose the alternative is only to look at the traffic flow characteristics in real time if and when there are connectivity issues - in order to solve those specific problems - and not store anything. One way or another, some level of access to the network data is needed if connectivity issues are to be solved, no?

Sebastian

**Comment by ajlennon** [comment]
@amcewen @paulfurley et al can we add an agenda item to any relevant upcoming meeting to come up with a statement of principle from DoES on how privacy is to be interpreted in terms of DoES users' communications (including that DoES presumably has some legal responsibilities on this nowdays?)

**Comment by ajlennon** [comment]
> One way or another, some level of access to the network data is needed if connectivity issues are to be solved, no?

I don't see that as required. no. Signal tests / IP connectivity tests can be carried out in isolation from users' usage I would have expected, or in the very worst case on a client-side opt-in basis

**Comment by xj25vm** [comment]
I was thinking of the specific case in which some users might complain that they have problems with their specific application - such as video conferencing for example. Without being able to look at the outgoing/incoming traffic for the whole site, work out which applications or protocols use the bandwidth and how - and then work on a solution involving QoS or some reconfiguration of existing network clients. This is just one example. I'm not sure how else would this be approached, if there is no access to some form of network statistics?

Sebastian Arcus

**Comment by zarino** [comment]
> @DoESsean @amcewen @paulfurley et al can we add an agenda item to any relevant upcoming meeting to come up with a statement of principle from DoES on how privacy is to be interpreted in terms of DoES users' communications (including that DoES presumably has some legal responsibilities on this nowdays?)

@ajlennon Maybe you should [create a new issue](https://github.com/DoESLiverpool/somebody-should/issues/new)? That way, at the very least, it’ll get picked up in the "let’s go through the issue tracker!" segment of Community Meetings.

**Comment by ajlennon** [comment]
@zarino I'm was trying not to proliferate issues. Maybe I can find the meeting notes from the last meeting and at it there was we seem to go through those next time around. What's best practise @DoESsean ?

@xj25vm I'm sure there are lots of edge cases for problems we haven't encountered yet. My own personal opinion is strongly that espoused by Stallman above. Any data collected will be misused (incuding what is termed 'meta-data') and thus data should not be collected unless absolutely necessary. Where determined to be absolutely necessary that collection should be done transparently giving individuals an opt-out

**Comment by de14n** [comment]
@ajlennon

I agree that Wireshark might be overkill, it is a beast. I was just looking up on what level of the OSI model it can work on, and it seems it can do up to level 2, leaving level 1, the physcial level left for interpretation. In which case you're looking at expert level RF analysing with a VNA or something insane like that.

Or we can just use the same kind of software you have pictured up there showing the strength of signals and their channels. You can get these type of software on mobiles too, might make it easier to scope the area for signal issues.

But if we're getting wired network then video conferencing should be fine. Wifi for mobiles and tablets.

**Comment by xj25vm** [comment]
@ajlennon - agreed with the idea of as little data collection as possible. Maybe even take the approach of designing the system as well as possible, and re-discuss ways of dealing with things if problems arise.

I was thinking that maybe QoS should be added to the original list of things to consider. VoIP and video conferencing depend on low latency - other protocols less so.

Also, maybe security should have its own little heading? Will there be devices internally on the network which will require more protection (as opposed to people bringing their own laptops in and taking care of their own firewalling etc.)? Or will the network be made up purely of client workstations - and everyone is responsible for their own?

Sebastian

**Comment by ajlennon** [comment]
@xj25vm I do agree re: QoS / and security is a big thing from a number of viewpoints.

We've seen what happened to the NHS recently when hit by WannaCry and I was half expecting to come in around then and find my systems on the WiFi had been encrypted so there's a huge potential risk there in future. Given we're all on the same network segment more or less we're all pretty exposed to whoever walks in off the street which is a concern.

Similarly I have ongoing issues with Skype which probably relate to some routing setting somewhere or other and which wold be nice to get to the bottom of one day...

**Comment by xj25vm** [comment]
@ajlennon - talking about security, some thinking might need to go into how accessible the wifi will be to guests, outsiders, visitors, attendees etc. Maybe even create a separate SSID on a separate VLAN for DoES'ers - on a slightly more secure network, and a separate one for visitors - to add some security layering and separation.

**Comment by ajlennon** [comment]
+1 for a guest network definitely!

**Comment by paulfurley** [comment]
OK I've tried to summarise that into what appears to be 3 new principles:

## Protect privacy by collecting as little data as possible

Any data collected will be misused (including what is termed 'meta-data') and thus data should not be collected unless absolutely necessary. Where determined to be absolutely necessary that collection should be done transparently giving individuals an opt-out

## Prioritise voice and video

VoIP and video conferencing depend on low latency - other protocols less so. We should consider implementing Quality of Service (QoS) to prioritise that traffic over low sensitivity traffic.

## Provide security through network segregation

If we're all on the same network segment more or less we're all pretty exposed to whoever walks in off the street.

We could provide a separate SSID on a separate VLAN for DoES'ers (or equivalent config for wired) - on a slightly more secure network, and a separate one for visitors - to add some security layering and separation.

We should do this in a way which makes visitors feel completely included and not some 'other'.

**Comment by xj25vm** [comment]
From a security point of view, it really depends on what else is/will be connected to the DoES network. If we are talking just about people using their own laptops, having only one SSID/network would mean you are no worse off than being connected to any hotel or café wifi. But if there are any internal file sharing services involved (not p2p, the 'old fashioned' type), or IoT devices - or anything else that might not have its own full firewall - then the case for a separate network makes sense.

**Comment by amcewen** [comment]
How many times does someone have to visit before they stop being a "visitor"?  (This is all excellent to watch going , I'm just throwing in the fact that there's less of a hard distinction between visitor and not-visitor in the DoES community :-)

**Comment by ajlennon** [comment]
Don't they pay to become a "member" ? :)

**Comment by SayBeano** [comment]
FWIW - Observium is just fine for traffic analysis. It's just collecting packets - no metadata in regards to websites or protocols. But it's fine in adaptive environments / finding a network hog / informing QoS.

It's runs fine and is speedy on a 4core with 2gig of RAM. It's got an 80GB disk and has over two years of history occupying 7gig of space for dozens of devices on a network that shifts on average a few TBs per month.

It knows what the devices know so there's no identifiable info. besides which port, router, modem, bridge, or switch did what.

![selection_007](https://user-images.githubusercontent.com/309325/32607442-fb900646-c550-11e7-97cf-a67a0411e975.png)

If you want to analyse at the level youse are talking (wireshark / tcpdump) about you'd do so adhoc there's no call to potentially enable DPI on a little network like this and the data you'd generate would be a sack of bricks. 

Observium can monitor servers / workstations too so you can have all your infra in it it can be a piece of mind as well as quality tool.

http://www.observium.org/screenshots/

**Comment by xj25vm** [comment]
It looks interesting - but unfortunately, based on their website, it looks like the useful bits - such as threshold and state alerts, and traffic accounting, are missing from the community version

**Comment by paulfurley** [comment]
@amcewen I suggest that we could set it so the more-segregated-network is opt-in, in that it's not the default. e.g. it could be a hidden SSID or something. 

Regarding monitoring tools, can I suggest we take that discussion to the [mailing list](https://groups.google.com/forum/#!forum/DoES-Liverpool) and keep this issue focused on principles for the network? Thanks!

**Comment by mattwilsondotuk** [comment]
Hi Guys, 

Quick update from me and what I'm doing in the building...

Access wise - I'm putting a 10GB pipe into the building straight from our DC as I'm building an Internet Exchange the other end of the building which will have FOC access onto IOT lan

I'm opening up 500MB for shared access between internal clients at the Tapestry to keep costs low for you guys and others, but I can open more if required in 1GB segments, just need a few hours notice.  It's not broadband, its synchronous internet, so same speed both ways (too much hassle to water down the beer!)...I can deliver this via fibre to the doES floor if you have a fibre switch, or cat6...your choice..John already has pricing on this...

IP Addresses - Can give you what you need, IPV4 or IPV6

WIFI - Down to Does but I would like to do something with this and supply say 3/4 AP's for 2G and 5G access

City wide network name - we are already doing, and if we do the WIFI inside, you can use the service outside around the fabric district (roam)...and many other locations around the city that we have service at...all for free..essentially taking some of your doES desk benefits with you..

Wired points, I suggest a good managed GB switch and can get a good price on these and will setup VLANS foc for network segregation

Monitoring - We will supply tools for free if we do the WIFI for monitoring speeds, signal from devices, blocking, setting QOS, guest access polices etc etc..

USB dongles - not required, just setup fat free SSID, we can traffic shape to suit 

NAT - I can supply firewall, but you don't have to use NAT, I can give every device an IPV6 address with direct access so just use firewall to port block?

Service Prioritisation - doES clients can prioritise certain sites or services, like Github, Facebook, yada yada...see more at https://www.balticbroadband.com/features/

**Comment by mattwilsondotuk** [comment]
Just another announcement from a meeting we had today with the landlord and the news about what the Internet Exchange will bring plus how we are making it a Baltic Broadband Enabled Building coupled with the exchange...

Having an Internet Exchange in the basement of the building brings many benefits from speed to direct connections to other networks world-wide that by-pass the Internet giving low latency and high speeds, helping applications like Virtual Reality, Financial Services,Cryptocurrencies and Autonomous Systems,

We are also installing ports so that residents of The Tapestry can access the following services by simply connecting up a cable to our point of presence in the basement.

Direct Ports (by-passing the Internet) to the following Cloud Providers:

Amazon Web Services
Google Cloud Platform
Salesforce.com
Oracle Cloud
IBM Bluemi
SAP
Alibaba Cloud
Webair
Telecoms Cloud
We are also an LINX IXP Reseller, meaning that we can connect Tapestry Tenants to other Internet Exchanges, such as:

London Internet Exchange(LINX) (tenants can peer directly with networks or with anyone on the root servers, see those networks at https://www.linx.net/tech-info-help/members-by-ip-asn)

Manchester Internet Exchange (IXManchester)

Private Peering sessions or any networks in Telehouse Campus or Manchester Data Centers

Please let me know if anyone is interested as I'm wanting to get commitment/intent from interested parties asap so I can plan a go live date for everyone.

**Comment by amcewen** [comment]
On a more practical note (more of an FYI for when anyone gets to installing things), there's a fair bit of Cat6 cable in the "might be useful for fit-out" area of the staging room, and also 6 double-gang (i.e. the size of a double mains socket) 4-way RJ45 sockets (that I scavenged from the staging room).

We might be able to get more elsewhere in the building if they're useful.  There's also assorted bit of white plastic trunking to run cables in, but I don't know if we want that or not?

**Comment by mattwilsondotuk** [comment]
@amcewen plus anything that DoES had left over ,they could donate to IX Liverpool...they are desperately looking for a load of IT gear see www.ixliverpool.net/contributions/

**Comment by DoESsean** [comment]
Have there been any further ideas about this? Or some movement on whatever needs moving on?

(As you may be able to tell, this is about a million miles over my head.)

**Comment by mattwilsondotuk** [comment]
Hi Sean, i agree I think we need to bottom out what is happening with the WIFI etc, as this will only cause delays when DoES are moving in and could be without service for some time.

**Comment by amcewen** [comment]
We need to work out exactly where the fixed network points are going to be in the new space.  @paulfurley, can you sit down with a copy of the floorplan sometime and spec out where the cables should be run, then we can liaise with the electrician to sort out getting them in.

**Comment by amcewen** [comment]
Any update on where any of the cables are going to go?  Where are the main patch panel and switches going?  Who is installing any WiFi access points?  Where are they being positioned?

I realise it's hard to say for definite without knowing exactly where the desks are being positioned, but it's getting to the point where we need to be making our best guesses...

This issue is also pretty long and wide-ranging, so it would be great if we can spin out a bunch of more focused issues and then close this one as the "planning" phase will be over.  If they all get the `DoES Move: The Tapestry` and `System: Network` labels then it'll still be easy to get an overview as well as the detail.

**Comment by johnmckerrell** [comment]
My hope has always been for minimal but I know @paulfurley has strongly suggested maximal! I agree that it's better to do stuff now than wait and have to run it in conduit unnecessarily. The main thing would be that we get at least 1 cable to each area, and then if we have 1, why not 2.. or 3??

Right now the electrician we're talking to has stated he does not do data. Perhaps we can do it ourselves before the stud work is fully boarded and while the electrician has any floorboards up, or I believe @mattwilsondotuk has someone we can ask to quote.

If anyone is taking on the project of defining where the cables need to terminate, the diagrams on the electric issue #664 should give good suggestions.

**Comment by johnmckerrell** [comment]
Also I've arranged to have a meeting with @mattwilsondotuk on the 12th but that's mainly to discuss Wi-Fi positioning, if someone wants to make strong suggestions before then feel free, and bear in mind that some of these cables will want to go high up for the access points.

**Comment by amcewen** [comment]
If @paulfurley wants something nearer to maximal then he should pull his finger out and start speccing where it should go ;-)

**Comment by mattwilsondotuk** [comment]
Yeah I know someone who can run some cables in for you for a good price. I would suggested some ceiling and wall mounted AP's (next to sockets) and fed by the conduit, assuming that is the wiring system that you are going for, so you will need to get the cables in like pronto before anything is boarded.

Be aware that we need to run such cables totally away from any cabling carrying heavy loads due to induction, which will cause interference, so care must be taken to route cabling away from electrical and other cables as part of the design.

Do you have architects drawings of the proposed cabling system and the specified routes being taken?

**Comment by paulfurley** [comment]
Arg sorry only just saw these updates from a week ago, for some reason.

I'm in London atm and abroad next week, but can join the meeting on the 12th if that helps?

**Comment by johnmckerrell** [comment]
Here's my starter-for-10. Big "N" is a required port, little "n" is more of a nice-to-have. This is IMHO, so anything can be changed, but at a minimum we do need wired connections in the event room, meeting room and soundproof room.

I haven't covered the access points largely as we don't know where they're going. I'm wondering if two back-to-back on the central RSJ would work.

![updated-tapestry-does-networks](https://user-images.githubusercontent.com/27243/36900648-049bfe9c-1e1c-11e8-8317-0869e4ff7f3a.png)

**Comment by magman2112** [comment]
I would tend to agree with the "Flood" wiring of Cat 6 cabling that has been mentioned above. It is possibly going to be used for telephones and possibly IoT devices with PoE and IP Camera's as well as LAN and internet, so for future flexibility we need to agree a wishlist then see what is practical.

It is quite a few years since I designed and installed any structured wiring, but the one practical thing to remember is that the valume of wires add up. The cable may only be 5-6mm nominal diameter, but can add up to quite a few square inches of volume at the hub (mixing my measurements, but I'm old enough to get away with it :-)).

I don't see a designated location for the Wiring closet as yet - all of this Cat 6 has to be terminated somewhere, along with LAN switches and possibly WLAN controllers. Somewhere in the Kitchenette may be favourite, being reasonable central to shorten the cable run lengths. We will also likey need a rack cabinet of some type, either floor or wall mounting - and will other equipment/servers want to live in this same rack? eBay is normally favourite for this kind of thing, especially if it can be picked up.

Another consideration is chosing the quality level of the installation, as this has a significant impact on cost. Such as metal trays under the floor for grouping the cables back to the wiring closet. Plastic trunking (with lids that invariably come off over time, or plastic conduit (with how many cables can you fit in one conduit being a consideration), again a cost v quality issue.

Last, but no means least, I'm happy to donate my time/tools to help with the installation if needed, but I doubt that I could undertake an installation on my own in a timely manner.

**Comment by amcewen** [comment]
As I mentioned in #680, it looks like the Internet will be terminating in the area between the kitchenette and the meeting nooks.  We should have this small server cabinet to house the patch panels and switches...

![39927291694_8038543312_k](https://user-images.githubusercontent.com/108468/37040370-933eb3fa-2151-11e8-94d9-cfb53318b550.jpg)

**Comment by xj25vm** [comment]
Sorry - I haven't had any update from this thread in at least a month, for some reason. Looking at the diagram above, it looks like quite a few x's to mark network points. The cabinet above is not that big. I still have lots of scars on my hands (and traumatic memories to go with them :-) ) to prove that stuffing and organizing even fewer cables into a cabinet of that size is no fun job. Also it is amazing how cabinets attract all sort of other bits of equipment in the long run - CCTV dvr's, UPS's, fibre/ADSL modems, separate PoE network switches, even the odd server or two. I have never been sorry for installing a cabinet which was bigger than needed :-)

**Comment by zarino** [comment]
> stuffing and organizing even fewer cables into a cabinet of that size is no fun job. Also it is amazing how cabinets attract all sort of other bits of equipment in the long run

If you need space for a server cabinet (or just shelves with switches/routers on them), it’s currently looking like there’ll be a storage large cupboard, roughly 1.8m wide, 0.9m deep, and up to 3m high, at the exact spot you’re talking about – between the kitchen and the meeting nooks. Access will be via standard full height doors incorporated into the "back" of the bench at the end of the meeting nook.

See the side-on plan in https://github.com/DoESLiverpool/somebody-should/issues/680 for more detail.

**Comment by amcewen** [comment]
@magman2112 and @ajlennon started working out cable runs yesterday, using @johnmckerrell's plan from above.  @magman2112 and Simon Derwent should be started on running actual cables today, and will be proceeding as quickly as possible to get it laid out.

As there's been some concern over the size of the small server cabinet that we liberated from next-door, @magman2112 suggested we take the rails out of the full-height server cabinet that we'll be using to house the Ultimakers; then we can cut them to whatever length we like and screw them to some studwork in the large cupboard.  So we'll need to work out exactly how that gets built into the cupboard, maybe as part of #680?

**Comment by ajlennon** [comment]
Yes we started looking at this yesterday. @magman2112 has this all in hand and we discussed what needs to be done to route the networking cables asap to unblock the builders and enable them to continue with finishing the partition walls. As of yesterday they had to stop to wait for us to finish this work and hopefully will be able to resume from tomorrow (Tuesday 13th)

I'm going to put my notes here and if I misrecollect anything I hope @magman2112 will correct me

- the electrics have already gone in mostly. There are plastic and metal housings on the partition walls with the power cables dropping down through floor boards and then bring routed along underneath the floor boards through holes which have been drilled in joists. Joists run at right angles to floor boards (for the uniniated like me) and there's an issue as quite a lot of holes have to be drilled through joists to get cables from A to B.

- we contacted the electrician about doing the work as we need a special joist drill to be able to to the job but ultimately the electrician felt he'd have to do at least half the work again to drill out everything for the networking cables. Thus this would cost us.

- @magman2112 is very much on the case with all of this and as such with @jackie1050 we decided to rent a "Joist Drill" from HSS for a couple of days to enable him to drill through the joists today to enable us to move forward. He's picking/picked up the drill from St. Helens this morning.

**Comment by ajlennon** [comment]
First off - slightly away from networking but we were looking at this panel by the entrance to the main area. This is where it has been suggested we put a sink. Apparently plumbers think this might be a problem as the soil pipe is a way away by the kitchenette area.

So number one: Do we want a sink here? If not do we have two sinks in the kitchenette area (clean versus dirty as in Gostins?) Or something else?

https://www.amazon.co.uk/photos/share/4T2ChRzH75VKQ4pAHabtj0AMA2LlHEtPSKvdNzzf11P

**Comment by zarino** [comment]
@ajlennon I’d suggest moving that to a new issue ticket, because it doesn’t have anything to do with planning excellent internet.

**Comment by ajlennon** [comment]
Is there a "planning excellent mains power" ticket or something similar? I don't want to start opening up random orphaned issues

**Comment by zarino** [comment]
I’m one of those people who has no problems creating tiny, single-topic tickets: https://github.com/DoESLiverpool/somebody-should/issues/686

**Comment by ajlennon** [comment]
CABLE ROUTING as of 12th March 

Mark has annotated on @johnmckerrell 's original network diagram where he thinks the routing needs to go. A lot of this is near where the electrician has already lifted boards and drilled through joists.

- This is on the basis we want to lift as few boards as possible as the floor is going to be left uncovered and we want it to look as good as possible and creak as little as possible

- @magman2112 also suggests we screw down the lifted floorboards to ensure they are secured and easily accessible in future.

- To minimise radiated interference we need to ensure the network cabling is not too close to the mains cabling where it parallels the cabling.

- The routes also try to minimise the amount of joists we have to drill through

- The two different routes into the patch panel from RHS and CENTRE/LHS are taken to avoid introducing structural weaknesses.

- We've introduced numbers to show the number of planned network points.

- This may need to be revised and we may need more network and power points in the quiet space. (@amcewen ?)

- The co-working space / event space has been planned out but not yet annotated (@magman2112 if you can get me a snapshot of the updated diagram I can go over in red here?

![cable routing](https://user-images.githubusercontent.com/1537834/37284144-5b042042-25f2-11e8-8953-289030496d4a.jpg)

**Comment by ajlennon** [comment]
OK - then can you ensure that ticket is acted upon - thanks @zarino

**Comment by ajlennon** [comment]
Does anybody know where we are with @mattwilsondotuk 's external Internet coming in?

Could we agree a tentative date for when we should be able to start connecting our network box up and moving packets around to the outside world?

**Comment by ajlennon** [comment]
I've followed your lead @zarino and broken out #687

**Comment by johnmckerrell** [comment]
The external internet feed depends on power being provided to their hardware which they're waiting for the landlord to provide, there will then be a period of potentially weeks. Talking to @mattwilsondotuk this morning it certainly looks like this could be a blocker to moving in on the 25th but it really depends on when he can get power to his hardware.

**Comment by ajlennon** [comment]
@johnmckerrell yikes. When will we have clarity on the date? How does failing to leave Gostins on w/e of 25th affect tenancy agreements and so forth? How difficult is the power issue? i.e. is it just plugging something in and switching it on or is it digging up a road to lay a cable? Is there any other way to support Internet in the short term as a plan B?

**Comment by magman2112** [comment]
Thanks to Simon and JR Peterson for their excellent help today with the network cabling. We just managed to meet our primary target of fully installing the 12 cables that releases the builder to board the remaining walls in the space.

A lot of the work today was prep work. Simon wore out one drill bit, due to the number of holes we needed to drill, plus we needed to lift quite a few floor boards to enable some runs and holes. Some drilling is still needed tomorrow, but we have now agreed to limit noise disturbance to other tenants between 10am and 3pm, I am planning to go into Tapestry early tomorrow to complete any noisy work that may limit the cables we can install tomorrow. Then we can concentrate on pulling in as many cables as possible.

Quite a few (20) of the remaining cable runs are over 22 metres in length, with the cable on hand now being shorter than this (we have already used all the longer lengths we could find). After discussing this with Jackie, I recommended that I pick up a box of Cat 6 cable tomorrow that will enable us to complete this part of the cabling. I expected this to cost circa £100 inc VAT for a 305m (1000ft) box from Screwfix.  I have just figured out though that we actually need approx 450m of cable for these 20 runs. From Screwfix, a 100m reel is £29.99, so 5 reels of 100m is £10 cheaper than 1 of 305m plus 2 of 100m, so this is actually better value for more cable. I will check this with John before going ahead though.

If anyone knows of any other long lengths of cable, please let me know before 10am, when I plan to go and purchase the new cable. New cabling is also about twice as fast to install compared to re-using cable.

**Comment by johnmckerrell** [comment]
@ajlennon we're ok in Gostins for longer if we need

**Comment by johnmckerrell** [comment]
FYI we've had a complaint about the noise of the fit-out from the ground floor tenants. They're being very nice about it but if you're working in the building please try to limit the noise you make between 10-3 Monday-Thursday and Saturday 9:30 - 4:30. Thanks!

**Comment by johnmckerrell** [comment]
We've had an offer of 3 free Unifi AP-AC-LR units from @skos-ninja which is great! Current thinking is that we use these for the coworking room, events room and the big main room and see how we get on in the meeting room (which will be well equipped for hard-wired networking anyway). I don't think we've finalised the position for the APs though so will still need to do that, and then wire in cables.

**Comment by magman2112** [comment]
Good progress was made on the network wiring today, with 21 cables being installed. I had excellent help from Chris Fairey today) apologies if I spelt Chris’s surname wrong as I got this via a telephone call with Jackie). I am sure we will both be feeling the workout we had today for a day or two.

If you are wondering about the odd one cable in the 21, this was a high level connection in the meeting room for a possible WLAN device.

With a few recent additions and assuming 4 points at most locations, the total cable count currently stands at 93 for the full installation. So far, 33 of these have been run. I have drawn up a clearer plan of the cable runs on paper today, which I plan to convert to an electronic format tomorrow hopefully.

I have other activities outside of DoES for the next few days, but plan to get back to this wiring on Saturday.

**Comment by magman2112** [comment]
One thing I have mentioned to both Jackie and John is the socket boxes on the brick walls.

The electrician has installed metal back boxes like these 

http://cpc.farnell.com/mk/k2214alm/box-2-gang-with-knockouts/dp/PL14730?st=Metalback%20boxes

From an aesthetic point of view, we should ideally use the same type of back boxes, along with black conduit up from the floor, but obviously the costs are mounting up for this installation. I suspect though that we may be able to get an equivalent back box cheaper than the one I listed above and I now I have found a LAN front plate to match the back box at a reasonable price.

https://www.sparksdirect.co.uk/BG-Nexus-4-Module-Rectangular-Metal-Clad-Front-Plate-100-50mm-Euro-Plate#.Wqj5t1qnyhB

These faceplates take Euro size modules, which I suspect the LAN sockets in the faceplates we have already got are the same size (and hopefully any more faceplates we recover from Gostins). I managed to find the same faceplate on eBay for £3.75 including postage, so I bought one to test this assumption.

Overall, we currently need 10 of these back boxes and faceplates for the network installation.

**Comment by jackie1050** [comment]
At close of play Sunday 18 March, all wiring in main room installed. Next is the installation of socket boxes and terminations, and cabling of coworking and event space.

We need to order (@magman2112 will order on Monday if no objections):

10 2 gang metal back boxes £3 each £30
2 1 gang metal back boxes £2.50 each £5
10 4 way Euro faceplates (metal) £3.75 each £37.50
2 2 way Euro faceplates (metal) £3 each £6
3 reels 100m cat 6E £30 each £90
93 cat 6 sockets c£2 each £186

Estimated additional expenditure £354

We still need to decide on the cabinet and how it’s going to be installed. @magman2112 suggests freestanding to allow easier access to power etc.

**Comment by mattwilsondotuk** [comment]
Just a reminder about the Internet, now we have power on the site as of yesterday (yipee!) we can now start to build the exchange and various other systems.  For Does, we need a network diagram, details of the firewall you are using (if any) and how you intend to terminate the connection into the office. 

I was at the site on Sat and ran in the Internet feed cable with one of the volunteers who made a good job of the cabling, taking it into the rack area ready.

Positions of the AP's and wiring for them has been confirmed with @johnmckerrell 

It would be good if your office firewall/router supported VLAN's so we can offer Layer 2 services up at Does, i.e direct connections to amazon, azure, IOT LAN's etc..

**Comment by skos-ninja** [comment]
@mattwilsondotuk after speaking with @johnmckerrell a fair bit I think we were intending to get a [Unifi Security Gateway](https://www.ubnt.com/unifi-routing/usg/) to allow us to do QoS and we should be able to setup multiple VLAN's fairly easily with this setup.

I'm currently working on contacting Ubiquiti to see if they can do anything to help sponsor this equipment and should hear back soon from them.

**Comment by johnmckerrell** [comment]
Thanks Jake. We don't currently have a network diagram. Networking's not my thing so I'd struggle to prepare one. We generally just block everything incoming apart from the webcams and the doorbots. I was also going to confirm what hardware we already have that we can continue to use. We definitely need some PoE gear as we have none currently. This issue is really holding out for a hero, people are giving great feedback but there's no one person leading this one.

**Comment by johnmckerrell** [comment]
Just checked and we have a HP Procurve E2510-24 Switch (J9019B) currently. It's a managed switch with gigabit ethernet. All of the other equipment we currently have is either not gigabit, noisy or desktop (i.e. 4-12 port type thing).

@mattwilsondotuk I thought I'd asked via email but don't now see it. Will the router you provide offer firewall capabilities, port forwarding, DHCP?

**Comment by mattwilsondotuk** [comment]
@johnmckerrell yes it will but only as part of a managed service, i.e you won't be able to access it from within Does as its a customer CPE so customers don't have access to it as standard. Normally customers connect their firewall to it.

**Comment by jackie1050** [comment]
Volunteer who helped @mattwilsondotuk on Sunday was JR (@thingomy)

**Comment by magman2112** [comment]
@johnmckerrell That switch should be plenty for the coworkers for the move, but ideally we should keep an eye out for 1 or 2 similar switches in the near future to increase the wired coverage at Tapestry. We may have to do a bit more patching in the short term, as people start to figure out how they are going to use the new space.

I suspect we also still need a PoE switch in the short term, to power the Wi-fi routers that have been donated.

**Comment by skos-ninja** [comment]
After discussing with @johnmckerrell we think for the moment getting 1 router and an 8/16 port poe switch will work best.

At the moment we are after using a Raspberry Pi as a Unifi Controller and then getting a Unifi Security Gateway and one of their Unifi PoE Switches to compliment the 3 Unifi AP's we already have to allow ease of management and overview of the network. We suspect the main usage of the internet will still be via WiFi so adding the existing HP Procurve E2510-24 should give us enough ports for now with options to upgrade in the future.

I have offered to @johnmckerrell to get a [US-16-150W](https://amazon.co.uk/dp/B01E46ATQ0) to help with this so DoES should only need to purchase a [UniFi Security Gateway](https://amazon.co.uk/dp/B00LV8YZLK) and find a spare Pi.

**Comment by xj25vm** [comment]
In case it helps (just ignore me if it doesn't):

1. I have a bunch of TP-Link gigabit 16 port PoE switches installed and running wifi points and cctv cameras for 5-6 years now - and they seem pretty solid to me - and cheaper than other brands (although they have gone up from £90 last year to nearly £150 now!). I have just received today in the post a replacement of the above model in warranty - after it started to go wobbly after 2.5 years of service. They have a lifetime warranty - and it's good that TP-Link honours it (after some wrangling over email with their returns department, in all fairness):

https://www.amazon.co.uk/TP-Link-TL-SG1016PE-16-Port-Gigabit-Switch/dp/B071ZGZ56M/

3. If PoE is not needed, a TP-Link unmanaged gigabit switch of 24 ports is only £67 at the moment. I run sites of up to 100 network points on unmanaged switches, and I think that, unless the features of managed switches are truly needed, they add a significant whack to the budget - and to the complexity of the setup. Many sites don't need managed switches, and don't really have the skills to use those feature, IMHO.

4. For a firewall and QoS, provided that the skills are available, a Linux box running on commodity hardware is hard to beat. The configurability and flexibility is almost endless, it can be replaced in a pinch for almost no cost if it goes bang and you are not tied to the standards or compatibility of a particular manufacturer.

OK - I'll get my coat now :-)

**Comment by skos-ninja** [comment]
My main thought with the router is that we need something that is going to be reliable and has customer support if we need it as internet is pretty much a requirement at all times at DoES we can't be having something that goes bang ;)

**Comment by mattwilsondotuk** [comment]
Just confirming that we've made 100MB upload and download live early on this week on the DoES  floor and on Monday 26th March the 500MB service will start after making quite a lot of progress this week on it.

For this we've had to forfeit static ip's, VLAN's the IX Liverpool exchange connection IOT network and Layer 2 services that will come on-line later in April now. but at least DoES has the "full fat" connectivity from Monday now :)

**Comment by skos-ninja** [comment]
A thing of note is the HP ProCurve 2510-24 seems to only support 1000Mbps on 2 ports so I'm not sure how useful this switch is.

**Comment by skos-ninja** [comment]
Just to note it down here. 

The three unifi ap's that had arrived have been setup and are ready to be plugged in via PoE injectors for now. 
Also we now have a raspberry pi setup running the UniFi controller software on https://192.168.1.2:8443 that is in the cabinet, do note that this pi is set to a static ip of 192.168.1.2 which the ap's are expecting for the controller software. This pi can be moved and plugged in else where if need be and is not required for the ap's to be working.

@johnmckerrell has ordered the other pieces of equipment and we hope it should arrive early this week. I'm happy to come by after work one day once it has arrived to help set this up too.

The login details for the UniFi software and the pi are currently on my laptop, if an organizer would like the details please contact me and I can forward them to you.

**Comment by Thingomy** [comment]
Current state of play is, wifi is up, and the 4 network points in event space are live, but most of it is still on 100Mb due to issues with the switch. All of this is somewhat temporary as things are still a bit up in the air.

Wifi coverage seems excelent from a brief walk around.

**Comment by skos-ninja** [comment]
One thing I have noticed is that the router @mattwilsondotuk has provided is doing DHCP but is locked to balticbroadband which means we can't do static ips right now as we are unable to reserve the addresses. Is there a way around this?

**Comment by Thingomy** [comment]
Last night all of the remaining network lines were put in bundles and fed up into the networking cupboard ready to be terminated by @magman2112 . Room side termination was done on the pillar in the middle of the main space.

**Comment by Thingomy** [comment]
Last night all of the remaining network lines were put in bundles and fed up into the networking cupboard ready to be terminated by @magman2112 . Room side termination was done on the pillar in the middle of the main space.

**Comment by magman2112** [comment]
The current state of play for the network wiring is that all of the accessible wall plates and sockets have now been installed and terminated (thanks to JR for his great efforts on this). The remaining terminations are:

- two floor boxes which are currently buried under flooring or furniture
- two wall mounted boxes between the front windows
- One socket high on the meeting room wall
- the connections in the quiet room, as the termination point has not been decided upon yet.

We need 4 more wall plates to complete this fit out, 1 metal clad and 3 plastic. I will order these later today.

In the wiring closet, 16 of the cables have been terminated so far. This covers all of the wi-fi and doorbot sockets and all of the sockets in the Events room. There is currently only one LAN connection patched through to the Events room, but more can be patched if required for Monday’s event. @johnmckerrell has ordered the patch panel we need to complete the termination in the wiring closet, this should be with us in the next few days, then we can look to complete the termination for all of the cables, ideally within the next week.

Wi-Fi is up and appears To be working well. I suspect some tuning is required though, as two of the devices appear to be using the same channel and @mattwilsondotuk has suggested that the nodes be set to a lower power mode.

There have been some discussions about LAN switches recently, we may need to discuss this further, especially deciding how many LAN ports that actually need to be patched in the space, along with whether VLAN’s are needed to separate co-workers from other potentially disruptive LAN traffic from the maker space, etc.

The only other aspect to completing this wiring is documentation. As much as possible, we are aiming to make the patching self documenting, by using logical naming and labelling standards. We will still need to generate some documentation, ideally electronically (especially for wiring runs, etc.). I asked @amcewen about drawing files for The Tapestry, as these would seem the most logical starting point for a wiring diagram. I will also build a spreadsheet for all of the wiring points to allow recording of usage as required.

**Comment by amcewen** [comment]
I've just dug out the SVG that I was talking about @magman2112, only to realise that it was before all of our fit-out, and so doesn't have any of the internal walls in it.

@zarino, maybe you could put the one you prepped [somewhere-safe](https://github.com/DoESLiverpool/somewhere-safe)?

**Comment by zarino** [comment]
@amcewen your wish is my command! https://github.com/DoESLiverpool/somewhere-safe/blob/master/does-tapestry-floor-plan.pdf

**Comment by johnmckerrell** [comment]
@magman2112 FYI the patch panel has arrived.

**Comment by skos-ninja** [comment]
At the end of play today I have now installed our Unifi Security Gateway.

The network is now setup to run on the 10.0.0.0/8 subnet.
All of the networking gear is given static ip's in 10.0.0.0/24.
Currently anything given DHCP address is given an ip in 10.0.29.0/24.

Again I still have the details for how to login which an organiser will need to grab off of me.

If a static IP is needed then someone with access to the unifi dashboard will be able to give them one.

I had a chat with @mattwilsondotuk about the network tonight in which we both agreed that DoES being given a static IPv4 and possibly a range for IPv6 would be our best bet as currently the network is getting triple NAT'd which isn't great. Matt has said it will come in the future and he will email DoES when he's ready to switch to a static IP.

Once we have a range for IPv6 we should allow people to request an IP with ports forwarded if they want for any little IoT device that they would like to access from the external network.

**Comment by ajlennon** [comment]
Fantastic news! Very keen that before any ports migth be forwarded internally we have a collective chat about VLANs, security and such?

**Comment by skos-ninja** [comment]
Yeah a security talk will need to happen before any port forwarding is done and even then it might be worth looking at separating the network more as currently everything is just sat on the same VLAN with no security around it which doesn't seem ideal.

Another thing that was mentioned by Julian yesterday was if we could give people a page in which they could find all devices on the network and their IP. I'm not sure how we would want to do this at the moment but seems like a good idea for all the Pi's!

**Comment by ajlennon** [comment]
"Julian yesterday was if we could give people a page in which they could find all devices on the network and their IP. I'm not sure how we would want to do this at the moment but seems like a good idea for all the Pi's!"

Absolutely. If we could do something around Bonjour/Zeroconf discovery and make that easily available to people I think that would be neat.

**Comment by skos-ninja** [comment]
Well with the joys of UniFi is that their dashboard already has this information available so we can just query this API to get most of the information required.

**Comment by ajlennon** [comment]
Awesome!

**Comment by skos-ninja** [comment]
I also wonder if it's worth designing a mount for our router, pi and baltic broadband's router to fit in a 1U as currently they aren't secured very well as they're just sat on top of one of the switches and it would tidy up the cabinet a bit.

**Comment by skos-ninja** [comment]
Yeah that does look good. 

I'm around Thursday evening to help @johnmckerrell with some doorbot setup and want to sort out #616 so you should see me hanging around the cupboard then.

**Comment by johnmckerrell** [comment]
This cupboard is an important bit of DoES infrastructure and shouldn't be considered fair game by all and sundry to start piling hardware into without having a conversation first about what you want to put in here. (Obviously @skos-ninja designing something to better host the various infrastructure hardware is a great idea)

**Comment by ajlennon** [comment]
I've been having a clearout in the garage and found a PoE switch. Presumably this would be useful in the networking cupboard @magman2112 @skos-ninja  ?

![poe box](https://user-images.githubusercontent.com/1537834/41201201-54d84874-6cab-11e8-88cb-c2c7abf44245.JPG)

**Comment by skos-ninja** [comment]
So during Maker Night last night me and @paulgeering made the mounting holes for the baltic broadband router (Edgerouter Lite), our UniFi Security Gateway and the Pi (Unifi Controller). 

This is now all mounted up and looks better and makes organising much easier now.

**Comment by johnmckerrell** [comment]
I think we can close this issue now, we've met the most important aims in @paulfurley 's origin comment, probably any further issues should be spun out as new issues. The main thing that's outstanding is the static IP but this is somewhat out of our hands.

**Comment by amcewen** [comment]
Adding a note here for next time I'm looking for it — @magman2112 has documented which network points go where in the wiring closet at https://github.com/DoESLiverpool/somewhere-safe/blob/master/Network/LAN%20Wiring%2016MAY2018.xlsx
